pkg/workloads/tf_api/api.py

```diff
@@ -89,18 +90,26 @@ def transform_sample(sample):
 
 def create_prediction_request(transformed_sample):
     ctx = local_cache["ctx"]
-
+    signatureDef = local_cache["metadata"]["signatureDef"]
+    util.log_pretty(signatureDef, indent=4)
+    signature_key = list(signatureDef.keys())[0]
     prediction_request = predict_pb2.PredictRequest()
     prediction_request.model_spec.name = "default"
-    prediction_request.model_spec.signature_name = list(
-        local_cache["metadata"]["signatureDef"].keys()
-    )[0]
+    prediction_request.model_spec.signature_name = signature_key
 
     for column_name, value in transformed_sample.items():
+        util.log_pretty(column_name, indent=4)
         data_type = tf_lib.CORTEX_TYPE_TO_TF_TYPE[ctx.columns[column_name]["type"]]
         shape = [1]
         if util.is_list(value):
-            shape = [len(value)]
+            shape = []
+            for dim in signatureDef[signature_key]["inputs"][column_name]["tensorShape"]["dim"]:
```

Comment:
Why is this reshaping necessary? Wouldn't `transform_tensorflow()` do the reshaping? We can talk about it if it's easier than responding here